test/tests/functional/pbs_reservations.py

```diff
@@ -883,3 +883,80 @@ def test_fail_confirm_resv_message(self):
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'},
+                            id=self.server.shortname)
+        now = int(time.time())
+        a = {'Resource_List.select': '2:ncpus=2',
+             'Resource_List.place': 'scatter',
+             'reserve_start': now + 30,
+             'reserve_end': now + 300}
+        r = Reservation(PBSROOT_USER, attrs=a)
+        rid = self.server.submit(r)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        resv_queue = rid.split(".")[0]
+
+        attrs = {ATTR_q: resv_queue, ATTR_J: '1-5',
+                 'Resource_List.select': '2:ncpus=1'}
+        j = Job(PBSROOT_USER)
+        j.set_sleep_time(60)
+        j.set_attributes(attrs)
+        jid = self.server.submit(j)
+        subjid = []
+        for i in range(1, 6):
+            subjid.append(j.create_subjob_id(jid, i))
+
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=30)
+        self.server.expect(JOB, {'job_state': 'B'}, id=jid)
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           extend='t', id=jid)
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           extend='t', id=jid)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[3])
+        self.server.deljob(subjid[3])
+        self.server.expect(JOB, {'job_state': 'X'}, id=subjid[3])
+        self.server.sigjob(jobid=subjid[0], signal="suspend")
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[2])
+        attrs = {ATTR_q: resv_queue, ATTR_J: '1-5',
+                 'Resource_List.place': 'scatter',
+                 'Resource_List.select': '2:ncpus=1'}
+        j1 = Job(PBSROOT_USER)
+        j1.set_sleep_time(60)
+        j1.set_attributes(attrs)
+        jid2 = self.server.submit(j1)
+        subjid2 = []
+        for i in range(1, 5):
+            subjid2.append(j.create_subjob_id(jid2, i))
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        self.server.sigjob(subjid[0], 'resume')
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid,
+                           extend='x', offset=120)
+        self.server.expect(JOB, {'job_state': 'B'}, id=jid2)
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           extend='t', id=jid2)
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           extend='t', id=jid2)
+        self.server.sigjob(jobid=subjid2[0], signal="suspend")
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid2[0])
+        self.server.sigjob(jobid=subjid2[1], signal="suspend")
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid2[1])
+        self.server.deljob(subjid2[1])
```

Comment:
Remove this line .
Instead of deleting  subjid2[1] , check subjid2[2] and subjid[3] started running.